macOS build fix

berendkleinhaneveld opened the pull request:
This MR fixes the build of pyopcode for macOS (Sierra, 10.12).

berendkleinhaneveld committed:
Don't redefine types on newer Apple systems

berendkleinhaneveld committed:
Update build number to adhere to meta.yml version

berendkleinhaneveld committed:
Apply some tricks from the VTK build script for configuring both the linux and osx builds

Korijn commented on conda-recipe/build.sh:
```diff
@@ -1,41 +1,38 @@
+    -Wno-dev \
+    -DCMAKE_BUILD_TYPE=${BUILD_CONFIG} \
+    -DCMAKE_CXX_FLAGS="-Wreturn-type" \
+    -DCMAKE_OSX_DEPLOYMENT_TARGET="10.12" \
```
Conda-build provides some env vars to maintain compatibility throughout the Anaconda ecosystem: http://conda.pydata.org/docs/building/environment-vars.html

In this case I think you should use `MACOSX_DEPLOYMENT_TARGET`; and be careful not to break the linux build where this envvar is unavailable :)

Korijn reviewed (changes requested):
Awesome! I have only one comment that needs addressing. I think we can merge afterwards.

berendkleinhaneveld committed:
Only add OSX deployment target on OSX and set to use minimal version 10.9

Conda hardcodes the MACOSX_DEPLOYMENT_TARGET value to 10.6, which
breaks the build and there is no way yet to specify a minimal version
of the SDK in any other way yet. For more details, please see:

https://github.com/conda/conda-build/issues/1269

Korijn commented:
Looks like our AppVeyor build is broken: I'm on it in #2 !
